packages/riipen-ui/src/components/Grid.test.jsx

```diff
@@ -0,0 +1,247 @@
+import toJson from "enzyme-to-json";
+import React from "react";
+
+import defaultTheme from "../themes/default";
+
+import Grid from "./Grid.jsx";
+import GridItem from "./GridItem.jsx";
+
+describe("<Grid>", () => {
+  it("renders without errors", () => {
+    let error;
+
+    try {
+      mount(<Grid />);
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toEqual(undefined);
+  });
+
+  it("renders correct snapshot", () => {
+    const wrapper = mount(<Grid />);
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  describe("default props", () => {
+    it("sets correct default props", () => {
+      const defaultProps = new Grid().type.defaultProps;
+
+      const wrapper = mount(<Grid />);
+
+      const component = wrapper.find("Grid");
+      expect(component.props().alignItems).toEqual(defaultProps.alignItems);
+      expect(component.props().justifyContent).toEqual(
+        defaultProps.justifyContent
+      );
+      expect(component.props().spacing).toEqual(defaultProps.spacing);
+      expect(component.props().flexGrow).toEqual(defaultProps.flexGrow);
+      expect(component.props().flexShrink).toEqual(defaultProps.flexShrink);
+      expect(component.props().flexDirection).toEqual(
+        defaultProps.flexDirection
+      );
+      expect(component.props().flexWrap).toEqual(defaultProps.flexWrap);
+      expect(component.props().onResize).toEqual(defaultProps.onResize);
+    });
+  });
+
+  describe("alignItems prop", () => {
+    it("sets given valid custom alignItems", () => {
+      const alignItems = "stretch";
+
+      const wrapper = mount(<Grid alignItems={alignItems} />);
+
+      expect(wrapper.find("JSXStyle").props().dynamic).toContain(alignItems);
+    });
+
+    it("gives an error given an invalid alignItems", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const alignItems = "middle";
+
+      mount(<Grid alignItems={alignItems} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("children prop", () => {
+    it("renders given non null children", () => {
+      const child = <GridItem />;
+
+      const wrapper = mount(
+        <Grid>
+          {null}
+          {child}
+        </Grid>
+      );
+
+      expect(
+        wrapper
+          .find("Grid")
+          .childAt(0)
+          .containsMatchingElement(child)
+      ).toEqual(true);
+    });
+
+    it("passes spacing to children nodes", () => {
+      const child = <GridItem />;
+      const spacing = 0;
+
+      const wrapper = mount(<Grid spacing={spacing}>{child}</Grid>);
+
+      expect(wrapper.find("GridItem").props().spacing).toEqual(spacing);
+    });
+
+    it("passes size to children nodes", () => {
+      const size = "sm";
+      const child = <GridItem />;
+
+      const wrapper = mount(<Grid>{child}</Grid>);
+
+      wrapper.find("Grid").setState({ size });
+
+      expect(wrapper.find("GridItem").props().size).toEqual(size);
+    });
+  });
+
```

Comment:
^I'm actually not a huge fan of that. That is testing specific implementation rather than testing input and o utputs.

I would rather see a test like:

1. Render the component
2. Simulate a resize event
3. Check to see if the new size is properly set or whatever it does

Reply:
The resize event simulated by Enzyme can't be detected by the event listener. I've also tried to see if we can use `ResizeObserver`, but in order to use it we'd have to mock all of the behaviour and that seems error prone to me.

Reply:
Ouuf! That's rough.

Ok let's go with the original PR implementation then :).